lib/weak_parameters/date_validator.rb

```diff
@@ -0,0 +1,40 @@
+module WeakParameters
+  class DateValidator < WeakParameters::BaseValidator
+    private
+
+    def parser_class
+      ::Date
+    end
+
+    def valid_type?
+      if options[:only]
+        Array(options[:only]).any? do |format|
+          begin
+            parser_class.strptime(value, format)
+            return false unless strictly?(format)
+            true
+          rescue ::ArgumentError
+            false
+          end
+        end
+      else
+        begin
+          parser_class.parse(value)
+          return false unless strictly?
+          true
+        rescue ::ArgumentError
+          false
+        end
+      end
+    end
+
+    def strictly?(format = nil)
+      result = format ? ::Date._strptime(value, format) : ::Date._strptime(value)
+      result.present? && !result.key?(:leftover)
```

Comment:
We can't use `Hash#present?` here because ActiveSupport is not loaded in this library.

Reply:
I'm so sorry.
refixed.